lib/auth/access_request_test.go

```diff
@@ -629,6 +632,152 @@ func testMultiAccessRequests(t *testing.T, testPack *accessRequestTestPack) {
 	}
 }
 
+// testResourceTTL verifies that the TTL for the Access Request gets reduced by
+// requested access time and lifetime of the requesting certificate.
+func testResourceTTL(t *testing.T, testPack *accessRequestTestPack) {
+	now := testPack.tlsServer.Clock().Now()
+
+	tests := []struct {
+		desc            string
+		expiry          time.Time
+		identityExpires time.Time
+		expectedTTL     time.Duration
+	}{
+		{
+			desc:            "access request with ttl",
+			expiry:          now.Add(8 * time.Hour),
+			identityExpires: now.Add(10 * time.Hour),
+			expectedTTL:     8 * time.Hour,
+		},
+		{
+			desc:            "access request without ttl",
+			expiry:          time.Time{},
+			identityExpires: now.Add(10 * time.Hour),
+			expectedTTL:     defaults.PendingAccessDuration,
+		},
+		{
+			desc:            "access request truncation by identity expiration",
+			expiry:          time.Time{},
+			identityExpires: now.Add(13 * time.Minute),
+			expectedTTL:     13 * time.Minute,
+		},
+	}
+
+	for _, tt := range tests {
+		r, err := services.NewAccessRequest("operator", "admins")
```

Comment:
I realize we're already in a subtest, so maybe this is a bad idea, but I would still probably put the body of this loop in a `t.Run`, otherwise it will be hard to tell which specific test is failing if one of these assertions fails.